prisma-fmt/src/hover.rs

```diff
@@ -0,0 +1,129 @@
+use log::info;
+use lsp_types::{Hover, HoverContents, HoverParams, MarkupContent, MarkupKind};
+use psl::{
+    error_tolerant_parse_configuration,
+    parser_database::ParserDatabase,
+    schema_ast::ast::{self, FieldPosition},
+    Diagnostics, SourceFile,
+};
+
+use crate::{offsets::position_to_offset, LSPContext};
+
+pub(super) type HoverContext<'a> = LSPContext<'a, HoverParams>;
+
+impl<'a> HoverContext<'a> {
+    pub(super) fn position(&self) -> Option<usize> {
+        let pos = self.params.text_document_position_params.position;
+        let initiating_doc = self.initiating_file_source();
+
+        position_to_offset(&pos, initiating_doc)
+    }
+}
+
+pub fn run(schema_files: Vec<(String, SourceFile)>, params: HoverParams) -> Hover {
+    let (_, config, _) = error_tolerant_parse_configuration(&schema_files);
+
+    let db = {
+        let mut diag = Diagnostics::new();
+        ParserDatabase::new(&schema_files, &mut diag)
+    };
+
+    let Some(initiating_file_id) = db.file_id(params.text_document_position_params.text_document.uri.as_str()) else {
+        info!("Initiating file name is not found in the schema");
+        panic!("Initiating file name is not found in the schema");
+    };
+
+    let ctx = HoverContext {
+        db: &db,
+        config: &config,
+        initiating_file_id,
+        params: &params,
+    };
+
+    hover(ctx)
+}
+
+fn hover(ctx: HoverContext<'_>) -> Hover {
+    let position = match ctx.position() {
+        Some(pos) => pos,
+        None => {
+            info!("Received a position outside of the document boundaries in CompletionParams");
+            panic!("Received a position outside of the document boundaries in CompletionParams")
+        }
+    };
+
+    let ast = ctx.db.ast(ctx.initiating_file_id);
+    let contents = match ast.find_at_position(position) {
+        psl::schema_ast::ast::SchemaPosition::TopLevel => format_hover_content("", "", ""),
+        psl::schema_ast::ast::SchemaPosition::Model(_model_id, model_position) => {
+            info!("We're inside a model");
+            info!("We are here: {:?}", model_position);
```

Comment:
Are these `info!` statements supposed to remain in `main`, or can they be removed? Do they appear in the standard output when using `tracing` or similar features?

Reply:
Like, I see them in the output tab of the VSCode video, but I don't understand their utility beyond testing that the function written here works

Reply:
+1

Reply:
Yeah, pretty much, they're not intended to stay :)